test_program/block.c

```diff
@@ -76,6 +96,8 @@ get_block(int pos)
 
         fstat(fd.fd_block, &st);
         size = st.st_size;
+        REQUIRE(size>0);
```

Comment:
REQUIRE is normally used to test the pre-conditions before starting the function.  The code should return an error or use assert() in line 99.